Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,423 @@
+        AZ::Entity* entity2 = nullptr;
+        AZ::EntityId entityId = CreateDefaultEditorEntity("ComponentModeEntity", &entity);
+        AZ::EntityId entityId2 = CreateDefaultEditorEntity("ComponentModeEntity2", &entity2);
+
+        entity->Deactivate();
+        entity2->Deactivate();
+        [[maybe_unused]] const AZ::Component* placeholder1 = entity->CreateComponent<PlaceholderEditorComponent>();
+        [[maybe_unused]] const AZ::Component* placeholder2 = entity2->CreateComponent<AnotherPlaceholderEditorComponent>();
+        entity->Activate();
+        entity2->Activate();
+
+        // When one entity is selected the component shows up like normal
+        AzToolsFramework::EntityIdList entityIds = { entityId };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 1);
+
+        // When both entities are selected, if there are no common components, the switcher is empty
+        entityIds = { entityId, entityId2 };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 0);
+    }
+
+    TEST_F(ComponentModeSwitcherTestFixture, DeselectingOneEntityWithMultipleEntitiesSelectedAddsRemovedComponents)
+    {
+        // Given two entities with different components
+        AZStd::shared_ptr<ComponentModeSwitcher> componentModeSwitcher = AZStd::make_shared<ComponentModeSwitcher>();
+
+        AzToolsFramework::EditorTransformComponentSelectionRequestBus::Event(
+            AzToolsFramework::GetEntityContextId(),
+            &AzToolsFramework::EditorTransformComponentSelectionRequestBus::Events::OverrideComponentModeSwitcher,
+            componentModeSwitcher);
+
+        AZ::Entity* entity = nullptr;
+        AZ::Entity* entity2 = nullptr;
+        AZ::EntityId entityId = CreateDefaultEditorEntity("ComponentModeEntity", &entity);
+        AZ::EntityId entityId2 = CreateDefaultEditorEntity("ComponentModeEntity2", &entity2);
+
+        entity->Deactivate();
+        entity2->Deactivate();
+        [[maybe_unused]] const AZ::Component* placeholder1 = entity->CreateComponent<PlaceholderEditorComponent>();
+        [[maybe_unused]] const AZ::Component* placeholder2 = entity2->CreateComponent<AnotherPlaceholderEditorComponent>();
+        entity->Activate();
+        entity2->Activate();
+
+        // When the both entities are selected, nothing shows up in the switcher as there are no common components
+        AzToolsFramework::EntityIdList entityIds = { entityId };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 1);
```

Comment:
`EXPECT_TRUE/EXPECT_FALSE` should only be used for boolean conditions.
If a relational operator(==, !=, <, etc...) is used, then those EXPECT_* macros should be used instead.
The reason why, is that when a test failure occurs, the relational `EXPECT_*/ASSERT_*` macros are able to provide resolve the values of both arguments and provide [better error messaging](http://google.github.io/googletest/reference/assertions.html#boolean).
If the EXPECT_TRUE` expression fails in this case the output would be "Expected componentModeSwitcher->GetComponentCount() == 1 is TRUE, actual FALSE"

Where as if the `EXPECT_EQ` expression was used, the output would be something like "Expected 1, Actual 3"

In this case `EXPECT_EQ` should be used
```suggestion
        EXPECT_TRUE(1, componentModeSwitcher->GetComponentCount());
```